src/array.jl

```diff
@@ -246,7 +246,7 @@ function convert(::Type{CategoricalArray{T, N, R}}, A::AbstractArray{S, N}) wher
     # if order is defined for level type, automatically apply it
     L = leveltype(res)
     if method_exists(isless, Tuple{L, L})
-        levels!(res, sort(levels(res)))
+        levels!(res.pool, sort(levels(res.pool)))
```

Comment:
Was this changed for efficiency as well? By calling `levels!` on the pools do you bypass safety checks that are slowing down the function?

Reply:
Yes, that's what I meant by "we do not actually need to call it from the constructor since we know the levels will all be preserved (just reordered)", though that wasn't really clear.